### ISSUE TITLE
Please provide example implementation for "getAuthTokenCallback"

### ISSUE THREAD

**Issue by sujayvsarma** [Azure-Samples/AzureMapsCodeSamples#19]
Hello,

When using the `authType` set to `anonymous`, we are required to provide a callback function that then provides the token. This method is heavily used on [this](https://docs.microsoft.com/en-in/azure/azure-maps/map-get-information-from-coordinate) documentation page.

However, there are no examples provided on what this called method is supposed to return. Calling the URI directly from my browser returns an OAuth token (ey...) but the documentation [here](https://docs.microsoft.com/en-us/javascript/api/azure-maps-control/atlas.authenticationoptions?view=azure-maps-typescript-latest) says [ **Properties** > `getToken` ]

> A callback to use with the anonymous authentication mechanism. This callback will be responsible for resolving to a authentication token. E.g. fetching a CORS protected token from an endpoint.

"CORS protected token".. So the question is, what is going on behind the scenes and how does one come up with a "CORS protected token" ?

**Comment by sujayvsarma** [comment]
Please see https://github.com/Azure-Samples/AzureMapsCodeSamples/issues/19 as well.

**Comment by rbrundritt** [comment]
I have asked one of our developers to look into this question. We will report back shortly. Also note that the [forums](https://stackoverflow.com/questions/tagged/azure-maps) or the [Azure support team](https://azure.microsoft.com/en-us/support/options/) is a much better place to ask these types of questions.

**Comment by walsehgal** [comment]
@sujayvsarma, the called function will configure the OAuth credentials to acquire an access token. You can visit [Azure Active Directory Authentication libraries](https://docs.microsoft.com/azure/active-directory/develop/active-directory-authentication-libraries) to take a look at the list of supported authentication libraries. Also the documentation on how to [manage authentication in Azure Maps](https://docs.microsoft.com/azure/azure-maps/how-to-manage-authentication) will help you understand on how to set up Azure AD app registration. Quick question, if you want to implement the function yourself what language do you want to use?

**Comment by sujayvsarma** [comment]
@walsehgal - 

I have already looked at the documentation you pointed to. They were either confusing or inadequate. I had raised a bug on them which they seem intent on closing instead of addressing. 

Which is why I raised this issue hoping that *developers* will help me with the problem. 

Language would be C#.

**Comment by rbrundritt** [comment]
Walsehgal is the doc writer for Azure maps.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by sujayvsarma** [comment]
OMG. Then this request is a waste of time already?

**Comment by rbrundritt** [comment]
You can also reach out to the Azure support team as well: https://azure.microsoft.com/support/options/ if you want more in-depth support.

**Comment by walsehgal** [comment]
@sujayvsarma, below is a sample async function deployed as an azure function, that acquires the access token. Hope this helps.

```aspx-csharp
public static async Task<IActionResult> Run(HttpRequest req, ILogger log)
{
    log.LogInformation("C# HTTP trigger function processed a request.");
  
    // Replace the tenant name 
    var authenticationContext = new AuthenticationContext("https://login.microsoftonline.com/<tenant_name>", false);
    
    // Config for OAuth client credentials 
    ClientCredential clientCred = new ClientCredential("<Replace with client ID>", "<Replace with client secret>");
    AuthenticationResult authenticationResult = await authenticationContext.AcquireTokenAsync("https://atlas.microsoft.com/ ", clientCred);
    string token = authenticationResult.AccessToken;
    return (ActionResult)new OkObjectResult(token);        
}
```

**Comment by sujayvsarma** [comment]
Instead of getting everyone in the world to implement this function, could you just open out the one Microsoft has already implemented - the one that is called in all your examples ? That would be way simpler.

**Comment by walsehgal** [comment]
This is the same function we are calling in our examples.

**Comment by sujayvsarma** [comment]
I am satisfied with the answer Ricky provided per [this reply](https://github.com/MicrosoftDocs/azure-docs/issues/38167#issuecomment-528517037). Specifically:

> A critical part of the service that you create is including your secret token. **We can't expose our service and make it reusable as there would not be a secure way to get your secret into it.** Also, a lot of companies add a lot of custom logic to their service and often run the service in the same domain as their application.

With that, as well as your code sample, I am happy to close out this issue.

**Comment by walsehgal** [comment]
Glad to hear that.

**Comment by sujayvsarma** [comment]
@walsehgal - One more question -- 

How do I correctly configure things between:

- Azure Maps Application (from Create Resource > Azure Maps)
- Azure App Service (that hosts the Maps stuff)
- Azure AD

Basically, how do I correctly register these with each other and in my Azure AD so that I can get a valid token? I tried to set it up from what I thought was logical, I ended up getting a token as well (from the code sample), BUT, Atlas Service API refused to honour the token (HTTP 403). So, I've done something wrong somewhere. But the terminology is far too confusing to attempt to resolve it.

Please help.

**Comment by walsehgal** [comment]
@sujayvsarma, you are assigning the role to a different service principal than the one you have the token for. Please make sure you get the token for the same service principle and then add it to RBAC.

**Comment by sujayvsarma** [comment]
@walsehgal - Basically here is what I went through:

First, I had only the Azure Maps Application (freshly created). Then from your sample code, I realized I need an App Secret. So I went to Azure AD and saw that (i) it was not registered there as an enterprise app and there was no way to do it. 

So, I created a new App Registration (Azure AD > App Registrations > New), set the URLs to my local development URLs (localhost stuff). I gave it `user_impersonate` delegated permissions to `Azure Maps` API (from API Permissions). Then, I generated a secret for this application. 

I used the Client ID and Secret from the App registration I had created to get the token using your sample code. It is for this token that I keep getting the HTTP 403 messages in the console logs. I can see that the call to the code is normal and seems to work fine -- My application is an ASP.NET Core 3 MVC application and I have implemented it as an Action in one of the Controllers instead of as a separate Azure function.
